tracking/tracking_tests/reporter_tests.cpp

```diff
@@ -0,0 +1,62 @@
+#include "base/thread.hpp"
+#include "coding/traffic.hpp"
+
+#include "platform/location.hpp"
+#include "platform/socket.hpp"
+
+#include "std/cmath.hpp"
+
+#include "testing/testing.hpp"
+
+#include "tracking/reporter.hpp"
+
+namespace
+{
+template <class Condition>
+bool WaitCondition(Condition condition, size_t durationMs = 1000)
```

Comment:
If you need to check that socket has output, why not to just set a callback in test socket, say, m_onOutput? And this callback will set all what you need and notify you via conditional variable?